elastalert/alerts.py

```diff
@@ -217,7 +214,13 @@ def alert(self, matches):
             to_addr = to_addr + self.rule['bcc']
 
         try:
-            self.smtp = SMTP(self.smtp_host)
+            if self.smtp_port:
+                self.smtp = SMTP(self.smtp_host, self.smtp_port)
+            else:
+                self.smtp = SMTP(self.smtp_host)
+            self.smtp.ehlo()
+            if self.smtp.has_extn('STARTTLS'):
+                self.smtp.starttls()
```

Comment:
You should send ehlo again after starttls (https://docs.python.org/2/library/smtplib.html#smtplib.SMTP.starttls)